### PR TITLE
Support parsing `ALTER STATISTICS` in PostgreSQL

### PR DESCRIPTION
**Pull request by everly-gif**
Ref #17848 .
Sub-issue of  #14015

Changes proposed in this pull request:
- Proofread `ALTER STATISTICS` grammar
- Support parsing `ALTER STATISTICS` in PostgreSQL
- Add tests

✅ Builds locally

### CODE REVIEW

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-postgresql/src/main/antlr4/imports/postgresql/DDLStatement.g4 [hunk 1]
```diff
@@ -1056,10 +1056,10 @@ foreignServerVersion
 
 alterStatistics
     : ALTER STATISTICS
-    ( ifExists? anyName SET STATISTICS signedIconst
```
Comment (strongduanmu) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-postgresql/src/main/antlr4/imports/postgresql/DDLStatement.g4:1059]: Why remove ifExists?

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-postgresql/src/main/antlr4/imports/postgresql/DDLStatement.g4 (hunk 1 above)
Comment (everly-gif) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-postgresql/src/main/antlr4/imports/postgresql/DDLStatement.g4:1059]: it doesn't exsits in the PostgreSQL dialect here 
https://www.postgresql.org/docs/current/sql-alterstatistics.html

Comment (strongduanmu) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-postgresql/src/main/antlr4/imports/postgresql/DDLStatement.g4:1059]: > it doesn't exsits in the PostgreSQL dialect here https://www.postgresql.org/docs/current/sql-alterstatistics.html

Can you check this grammar in PostgreSQL source code?

Comment (everly-gif) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-postgresql/src/main/antlr4/imports/postgresql/DDLStatement.g4:1059]: you mean the postgres **gram.y** file right?

```
AlterStatsStmt:
			ALTER STATISTICS any_name SET STATISTICS SignedIconst
				{
					AlterStatsStmt *n = makeNode(AlterStatsStmt);

					n->defnames = $3;
					n->missing_ok = false;
					n->stxstattarget = $6;
					$$ = (Node *) n;
				}
			| ALTER STATISTICS IF_P EXISTS any_name SET STATISTICS SignedIconst
				{
					AlterStatsStmt *n = makeNode(AlterStatsStmt);

					n->defnames = $5;
					n->missing_ok = true;
					n->stxstattarget = $8;
					$$ = (Node *) n;
				}
			;
```
There if exists is supported.

Comment (strongduanmu) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-postgresql/src/main/antlr4/imports/postgresql/DDLStatement.g4:1059]: Yes, document is sometimes inaccurate, and source code can give us better guidance.

Comment (everly-gif) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-postgresql/src/main/antlr4/imports/postgresql/DDLStatement.g4:1059]: thank you for this tip! will keep this in my mind for my next PRs . Ill revert the changes for `DDLStatement.g4`

**Review by strongduanmu — APPROVED** [review]
@everly-gif Another perfect pr. 👍

**Pull request merged**